includes/html/pages/device/apps/sagan.inc.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+use App\Models\Application;
+
+$sagan_instances = Application::find($app['app_id'])->data['instances'];
+
+sort($sagan_instances);
+
+$link_array = [
+    'page'   => 'device',
+    'device' => $device['device_id'],
+    'tab'    => 'apps',
+    'app'    => 'sagan',
+];
+
+print_optionbar_start();
+
+echo generate_link('Totals', $link_array);
+echo '| Instances:';
+$int_int = 0;
+while (isset($sagan_instances[$int_int])) {
```

Comment:
Replace this while loop with a foreach as I did in the other applications that used it.

Reply:
Done! And more.